lib/ace/ext/papeeria/latex_behaviour.coffee

```diff
@@ -0,0 +1,226 @@
+define((require, exports, module) ->
+  Behaviour = require("ace/mode/behaviour").Behaviour
+  COMMENT_TYPE = "comment"
+  ESCAPE_TYPE = "escape"
+  KEYWORD_TYPE = "keyword"
+  STORAGE_TYPE = "storage"
+  MATH_TYPE = "math"
+  EQUATION_TYPE = "equation"
+  RPAREN_TYPE = "rparen"
+  CORRESPONDING_CLOSING = {
+      '(': ')'
+      '[': ']'
+      '{': '}'
+  }
+
+
+  # If we're on the start of a line, but the current token is of a type "comment", we're not actually
+  # inside a comment, just on a start of it. That's the reason for `column != 0` check
+  isCommentToken = (token, column) -> column != 0 and token? and token.type.indexOf(COMMENT_TYPE) > -1
+
+
+  isEscapedInsertion = (token, column) -> (
+      token? and
+      (
+          token.type.indexOf(ESCAPE_TYPE) > -1 or
+          token.type.indexOf(KEYWORD_TYPE) > -1 or
+          token.type.indexOf(STORAGE_TYPE) > -1
+      ) and
+      column - token.start == 1
+  )
+
+
+  isInEquation = (session, row, column) ->
+    token = session.getTokenAt(row, column)
+    nextToken = session.getTokenAt(row, column + 1)
+    state = session.getState(row)
+    pState = session.getState(row - 1)
+    lastState = if typeof(state) == "string" then state else state[state.length - 1]
+    lastPrevState = if typeof(pState) == "string" then pState else pState[pState.length - 1]
+    return (
+        # This handles the case, when the cursor is on the empty string
+        (not token? and lastPrevState.indexOf(MATH_TYPE) > -1) or
+        # This handles the common case, when the cursor is inside the equation
+        (token? and token.type.indexOf(EQUATION_TYPE) > -1) or
+        # This handles the specific case, when the cursor is on the very start of the equation
+        (nextToken? and nextToken.type.indexOf(EQUATION_TYPE) > -1)
+    )
+
+
+  DO_NOTHING = null
+  AUTO_INSERT = { text: "$$", selection: [1, 1] }
+  SKIP = { text: "", selection: [1, 1] }
+  dollarsInsertionAction = (state, action, editor, session, text) ->
+    if text == '$' and not editor.inMultiSelectMode
```

Comment:
Looks like we return undefined if editor is in multi select mode. It is explicitly checked in `dollarsDeletionAction`, maybe do the same here?